src/main/Ui.java

```diff
@@ -27,37 +26,33 @@ public void dibujar(Campo campo) {
 			addKeyListener(campo);
 		}
 	    
-    	ubicacionesSerpientes.addAll(campo.notificarUbicacionesSerpientes());
-        ubicacionesComestibles.addAll(campo.notificarUbicacionesComestibles());
+		aDibujar.addAll(campo.notificarDibujables());
         repaint();
 	}
 
     public void paint(Graphics g){
         fondoDefault = new ImageIcon(fondoPath);
         fondoDefault.paintIcon(this, g, 0, 0);
-        if (!ubicacionesSerpientes.isEmpty()) {
-            pintarSerpientes(g);
-        }
-        if (!ubicacionesComestibles.isEmpty()) {
-            pintarComestibles(g);
+        if (!aDibujar.isEmpty()) {
+            dibujar(g);
         }
         g.dispose();
     }
 
-    private void pintarSerpientes(Graphics g) {
-        g.setColor(Color.BLUE); //TODO VER SI USAMOS IMAGENES EN VEZ DE CIRCULOS
-        for (Ubicacion serpiente : ubicacionesSerpientes){
-            g.fillOval(serpiente.getX(), serpiente.getY(), 20, 20);
-        }
-        ubicacionesSerpientes.clear();
-    }
-
-    private void pintarComestibles(Graphics g) {
-        g.setColor(Color.RED); //TODO VER SI USAMOS IMAGENES EN VEZ DE CIRCULOS
-        for (Ubicacion comestible : ubicacionesComestibles) {
-            g.fillOval(comestible.getX(), comestible.getY(), 20, 20);
-        }
-        ubicacionesComestibles.clear();
+    private void dibujar(Graphics g) {
+    	for (Dibujable dibujable : aDibujar) {
+    		List<Ubicacion> ubicaciones = dibujable.obtenerZonaDeDibujo();
+    		if (!ubicaciones.isEmpty()) {
+    			Color colorActual = dibujable.obtenerColor();
+        		g.setColor(colorActual.darker());
+        		g.fillOval(ubicaciones.get(0).getX(), ubicaciones.get(0).getY(), 20, 20);
+        		for (int i = 1; i < ubicaciones.size(); i++) {
```

Comment:
no se podría primero dibujar toda la serpiente (salteando la cabeza) y después en otro for dibujar las cabezas?
Lo digo para evitar el anidar fors, aunque no sé si le bajaría mucho la complejidad.